src/ast/scan.rs

```diff
@@ -6,6 +6,7 @@ const RADIX: u32 = 10;
 
 pub struct Scanner<'a, 'b> {
```

Comment:
see #28

Reply:
Indeed, this PR is not focused on the scanner, however.
I agree that we should add docstrings progressively as we improve the compiler, but I think that each PR should be somewhat focused on a single aspect.